packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -639,7 +626,7 @@ export function getValidatorApi(
       ? produceEngineFullBlockOrContents(slot, randaoReveal, graffitiBytes, {
           feeRecipient,
           strictFeeRecipientCheck,
-          commonBlockBody,
+          commonBlockBodyPromise,
```

Comment:
I think part of the issue we see below is `commonBlockBodyPromise` and `blockProductionRacePromise` are 2 separate promises
while we try to constraint the order of engine/builder vs `commonBlockBodyPromise` by passing it, it does not affect `blockProductionRacePromise` 
`blockProductionRacePromise`  always get started regardless